File: Businesscenterservices-web/src/app/shared/Model/CentreHospitalier.ts


import { Adresse } from "./Adresse";

export class CentreHospitalier {
  public id?: number; 
  public nom_ch?: string;
  public siret?: string;
  public email_ch?: string;
  public telephone?: string;
  public adresse?: Adresse;
  public adresse_id?: number; 
  public archived?: number; 
}

File: Businesscenterservices-microservices/centres-service/src/main/java/com/businesscenterservices/centres/entity/CentreHospitalier.java
package com.businesscenterservices.centres.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class CentreHospitalier {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String siret;

    @Column(nullable = false)
    private String nomCh;

    @Column(nullable = false)
    private String emailCh;

    @Column(nullable = false)
    private String telephone;

    @Column(nullable = false)
    private boolean archived = false;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "adresse_id")
    private Adresse adresse;
}
